### ISSUE TITLE
Question about trident loss

### ISSUE THREAD

**Issue by hezhu1996** [hudingding/TridentNet-mmdetection#4]
Hello你好~ 程序写的很清楚，我有一个问题关于loss就是，我理解的是trident在backbone就分成了3个branch每个里面都有rpn和rcnn然后对应一个loss，结果应该是3个branch的loss相加。 我看您和detectron2里面的实现在最后一个block中都是用了torch.cat把他们合起来，那之后的rpn和rcnn方面不用做改动嘛，loss也不用加起来嘛。。。 我没看明白呀 望解答~

**Comment by liqiang1127** [comment]
@TWDH 代码里面是每次随机选取3个分支中的一个送到下一步继续检测，官方实现是在batch_size纬度堆叠，然后对gt_bboxes和gt_labels都做了复制（复制三份），相当于后面流程1张图变3张图。

**Comment by hezhu1996** [comment]
谢谢回复~ 我理解官方代码是训练时三个分支同时进行的，而test时是用中间的分支预测， 不知道您这个随机选取是什么意思呢？ 还有个小问题，这三部分concat起来应该就相当于loss相加了吧，是不是单纯的除以3就是正常的loss呢， 灰常感谢~

**Comment by hudingding** [comment]
@TWDH ，如果想测试时只在单分支上预测，就需要将所有的gt，送入每个分支上进行训练，也就是去除scale-aware，此时有两个思路，1种就是将计算三个分支loss然后除3，保证和原始的网络loss差不多，这样学习率不用动；2种就是每次随机选取一个尺度，计算loss，这样loss和原来的网络loss级别差不多，而且只要训练的轮次多一点，效果应该是一样的，我是这么认为的
